File: tower.c
#include <stdio.h>

void tower_of_hanoi(int n, char source, char auxiliary, char target) {
    if (n == 1) {
        printf("Move disk 1 from rod %c to rod %c\n", source, target);
        return;
    }

    // Move n-1 disks from source to auxiliary
    tower_of_hanoi(n - 1, source, target, auxiliary);

    // Move nth disk from source to target
    printf("Move disk %d from rod %c to rod %c\n", n, source, target);

    // Move n-1 disks from auxiliary to target
    tower_of_hanoi(n - 1, auxiliary, source, target);
}

int main() {
    int n = 10; // Number of disks
    tower_of_hanoi(n, 'A', 'B', 'C');
    return 0;
}

File: tower.py
def tower_of_hanoi_recursive(n, source, auxiliary, target):
    if n == 1:
        print(f"Move disk 1 from rod {source} to rod {target}")
        return
    tower_of_hanoi_recursive(n-1, source, target, auxiliary)
    print(f"Move disk {n} from rod {source} to rod {target}")
    tower_of_hanoi_recursive(n-1, auxiliary, source, target)

tower_of_hanoi_recursive(3, 'A', 'B', 'C')